src/__tests__/config.js

```diff
@@ -0,0 +1,60 @@
+import {configure, getConfig} from '../config'
+
+describe('configuration API', () => {
+  let originalConfig
+  beforeEach(() => {
+    // Grab the existing configuration so we can restore
+    // it at the end of the test
+    configure(existingConfig => {
+      originalConfig = existingConfig
+      // Don't change the existing config
+      return {}
+    })
+  })
+  afterEach(() => {
+    configure(originalConfig)
+  })
+
+  beforeEach(() => {
+    configure({other: 123})
+  })
+
+  describe('getConfig', () => {
+    test('returns existing configuration', () => {
+      const conf = getConfig()
+      expect(conf.testIdAttribute).toEqual('data-testid')
+    })
+  })
+
+  describe('configure', () => {
+    test('merges a delta rather than replacing the whole config', () => {
+      configure({newKey: 123})
+      const conf = getConfig()
+      expect(conf).toMatchObject({
+        newKey: 123,
+        testIdAttribute: 'data-testid',
+      })
+    })
+
+    test('overrides existing values', () => {
+      configure({testIdAttribute: 'new-id'})
+      const conf = getConfig()
+      expect(conf.testIdAttribute).toEqual('new-id')
+    })
+
+    test('passes existing config out to config function', () => {
+      // Create a new config key based on the value of an existing one
+      configure(existingConfig => ({
+        newKey: `${existingConfig.testIdAttribute}-derived`,
```

Comment:
Rather than `newKey` in this test, we could set this to the `testIdAttribute` and get the same value out of the test without asserting on a non-feature.

Reply:
Yup, will update tests so they don't create random properties.  Would have been easier to test had there been _two_ supported properties from the start.